murfi-rt-PyProject/scripts/feedback.sh

```diff
@@ -519,4 +519,58 @@ then
     rm -f $subj_dir/rest/*bold.nii.gz
     rm -f $subj_dir/rest/*bold_mcflirt.nii.gz
     rm -f $subj_dir/rest/*bold_mcflirt_masked.nii.gz
+fi
+
+
+# ONLY if regular mask generation isn't possible
+# As a backup, we can register the template masks to 2vol space
+if [ ${step} = backup_reg_mni_masks_to_2vol ]
+then
+    clear
+    mni_template=MNI152_T1_2mm_LPS_brain.nii.gz
+    dmn_mni=DMNax_brainmaskero2_lps.nii.gz
+    cen_mni=CENa_brainmaskero2_lps.nii.gz
+
+    two_vol_ref=$(ls -t $subj_dir/xfm/series*.nii | head -n1)
+    two_vol_ref="${two_vol_ref::-4}"
+
+    two_vol_ref_bet=${subj_dir}/xfm/two_vol_ref_bet.nii.gz
+    two_vol_ref2mni=${subj_dir}/xfm/two_vol_ref2mni.nii.gz
+    two_vol_ref2mni_mat=${subj_dir}/xfm/two_vol_ref2mni.mat
+    mni2_two_vol_ref_mat=${subj_dir}/xfm/mni2_two_vol_ref.mat
+
+    study_ref=${subj_dir}/xfm/study_ref.nii
+    # if localizer_ref images doesn't exist yet, make it
+    if [ ! -f ${subj_dir}/xfm/localizer_ref.nii ]
+    then
+        mv ${study_ref} ${subj_dir}/xfm/localizer_ref.nii
+    fi
+    echo "Registering masks to reference image from most recent series: ${two_vol_ref}"
+    echo "study_ref.nii is now ${two_vol_ref}"
+
+    # Move the latest reference image to be study_ref
+    # study_ref.nii is used by MURFI to register to 1st volume of feedback runs
+    # So ROI masks need to be in the same space as study_ref.nii
+    cp ${two_vol_ref}.nii ${study_ref}
+
+
+    # skullstrip 2vol before registration
+    bet ${two_vol_ref} ${two_vol_ref_bet} -R -f 0.4 -g 0 -m # changed from -f 0.6
+
+
+    # first, register the 2vol to mni
+    # then calculate the inverse of the registration
+    flirt -in ${two_vol_ref_bet} -ref ${mni_template} -out ${two_vol_ref2mni} -omat ${two_vol_ref2mni_mat}
```

Comment:
Probably nbd, but is there a reason to register in this direction then invert instead of registering in the other direction and using the matrix directly?

Reply:
I didn't test this thoroughly, but in a few cases I thought I saw better registrations when using the MNI template as the reference, rather than going the other direction.